packages/xstate-test/src/machine.ts

```diff
@@ -64,7 +65,11 @@ function serializeMachineTransition(
     return '';
   }
 
-  return ` via ${serializeEvent(state.event)}`;
+  const prevStateString = prevState
+    ? `from ${simpleStringify(prevState.value)}`
```

Comment:
The solution starts here - we want to distinguish this transition from other transitions that may have the same target state and event by saying where this one came from